src/functional/common.py

```diff
@@ -74,6 +76,12 @@ def generate_operator(self, ir):
 class Connectivity(Protocol):
     max_neighbors: int
     has_skip_values: bool
+    origin_axis: Dimension
+
+    def mapped_index(
+        self, cur_index: int | np.integer, neigh_index: int | np.integer
+    ) -> Optional[int | np.integer]:
+        """Return neighbor index."""
```

Comment:
we should split this into static part encoded in the type and the runtime part, i.e. the function, but I'll leave that for later (parts are designed in the embedded field view prototype)